lab-will/server.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client.js');
+// const PORT = process.env.PORT || 3000;
+
+// Random Port;
+const PORT = Math.floor(Math.random() * ((9999-1000) + 1) + 1000);
+
+const server = net.createServer();
+const ee = new EE();
+
+const pool = [];
+
+ee.on('@all', function(client, string) {
+  pool.forEach(c => {
+    c.socket.write(`${client.nickname}:` + string)
+  });
+});
+
+ee.on('@dm', function(client, string) {
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').slice(1).join(' ').trim();
+
+  pool.forEach(c => {
+    console.log('client:', c.nickname);
+    console.log('nickname:', nickname);
+
+    if (c.nickname === nickname) {
+      c.socket.write(`${client.nickname}: ${message}`);
+    }
+  });
+});
+
+ee.on('@newname', function(client, string) {
+  let newName = string.split(' ').shift().trim();
+  client.nickname = newName;
+});
+
+ee.on('@close', function(client) {
```

Comment:
Instead of looking for the user to type @close you should be watching for the close event.  It is built in, similar to the 'data' or 'end' event.